cmd/testserver/main.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 Cloudflare, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A simple Roughtime server, intended for testing.
+package main
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"flag"
+	"log"
+	"net"
+	"time"
+
+	"github.com/cloudflare/roughtime/protocol"
+
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	// Command line parameters
+	addr           = flag.String("addr", "127.0.0.1:2002", "address to listen on")
+	rootKeySeedHex = flag.String("root-key", "", "hex-encoded root key seed (random 32 bytes)")
+
+	// Roughtime radius to use for responses
+	radius = time.Duration(1) * time.Second
+)
+
+func main() {
+	var (
+		err         error
+		rootKeySeed []byte
+	)
+
+	log.SetFlags(log.Lshortfile &^ (log.Ldate | log.Ltime))
+	flag.Parse()
+
+	// Set up root key
+	if *rootKeySeedHex != "" {
+		rootKeySeed, err = hex.DecodeString(*rootKeySeedHex)
+		if err != nil {
+			log.Fatalf("failed to parse root key seed: %v", err)
+		}
+		if len(rootKeySeed) != 32 {
+			log.Fatalf("unexpected root key seed length: got %d; want 32", len(rootKeySeed))
+		}
+	} else {
+		rootKeySeed = make([]byte, 32)
+		if _, err = rand.Read(rootKeySeed); err != nil {
+			log.Fatalf("rand.Read() failed: %v", err)
+		}
+	}
+	rootSK := ed25519.NewKeyFromSeed(rootKeySeed)
+	log.Printf("root public key: %s", base64.StdEncoding.EncodeToString(rootSK.Public().(ed25519.PublicKey)))
+
+	netAddr, err := net.ResolveUDPAddr("udp", *addr)
+	if err != nil {
+		log.Fatalf("could not resolve %s: %v", netAddr, err)
+	}
+	conn, err := net.ListenUDP("udp", netAddr)
+	if err != nil {
+		log.Fatalf("could not listen on %s: %v", *addr, err)
+	}
+
+	onlinePK, onlineSK, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		log.Fatalf("could not generate key: %v", err)
+	}
+
+	now := time.Now()
+	yesterday := now.Add(-24 * time.Hour)
+	tomorrow := now.Add(24 * time.Hour)
+	onlineCert, err := protocol.NewCertificate(yesterday, tomorrow, onlinePK, rootSK)
+	if err != nil {
+		log.Fatalf("could not generate certificate: %v", err)
+	}
+
+	buf := make([]byte, 1280)
```

Comment:
Any reason this is 1280 instead of 1024?

Reply:
@wbl may have insight here - this buffer size should be the same as in his original code.

Note that 1024 is the minimum request size, but the request is allowed to be bigger. I'm not sure why 1280 is picked but it's probably sufficient for testing purposes.